### ISSUE TITLE
Holding down tiltwheel on the Logitech G502 results in 1 second delay before performing action

### ISSUE THREAD

**Issue by rieje** [libratbag/piper#762]
I recently got a Logitech G502 and am using it on Linux. I use tiltwheel left/right for volume control. When I hold tiltwheel left/right, it will have a ~1 second delay before triggering the first action and then repeating it at a more reasonable rate until I release the button. If I were to spam click/release tiltwheel left/right, then the action is performed immediately. So there appears to be artificial delay when holding down tiltwheel. In practice, this means if I were to change volume by holding the tiltwheel left/right, there is a 1 second delay before anything happens. As a workaround, I would need to spam click the tiltwheel to change the volume in real time.

Is anyone experiencing the same problem on the G502? On my Logitech G9, I don't have this problem (holding down tiltwheel does not have a noticeable delay and the first action triggers immediately).

I have never used any Logitech-related software on Windows. Someone mentioned on Windows/MacOS, the key delay can be configured for the mouse, but I'm not even sure if this is a key delay-related issue because the first action is delayed (with key delay, the first action typically triggers immediately as you would expect and then the key delay takes effect).

I don't know where else to ask this, consider it a feature request. I've asked on IRC and subreddits.

**Comment by Summertime** [comment]
does this happen when bound to other keys? e.g. binding tilt to W

**Comment by rieje** [comment]
No, the delay is noticeably shorter, but I think it's because I've set the repeat_delay and repeat_rate for my keyboard strokes to be low and it's using that (feels like same delays when I compare binding tilt to W and holding W on keyboard). I guess default binding of tiltwheel left/right does not respect this for whatever reason.

**Comment by LeviVanDerMaas** [comment]
_TL;DR there is probably no easy way to fix this as I suspect it is hard-coded into the firmware_

Sorry for necroing, but to anyone else who's encountering the same problem and stumbling across this issue:

It does not seem to be an issue or missing feature within Piper specifically. On Windows, if you use the mouse while Logitech Ghub is running in the background, then tilting the wheel left or right will indeed emit a signal instantly (regardless of whether you are holding it or "flicking" it). However, if Logitech Ghub is **not** running, then even on Windows tilting left or right will introduce this artifical delay if you have horizontal scrolling assigned to the mouse-wheel tilts in the onboard memory.

From my own testing it also seems that when one assigns a keystroke (like "w"; e.g. something that'd normally come from your keyboard), this delay is not there. As such it seems to me that this delay on emission of a horizontal scrolling signal on wheel tilt is completely artificial and, for whatever reason, coded as a special case in the mouse's firmware itself (especially considering Ghub *can* detect a "held tilt" instantly even when it has horizontal scrolling assigned to it). 

My guess is such an artificial delay may have been initially implemented as a means to get a similar level of precision over scroll distance as one gets on the vertical axis. On the vertical axis you can precisely control your scroll distance as each tick of the scroll wheel increments the distance by a set amount, and because you can physically rotate the wheel it's very easy to scroll large distances. However, on the horizontal axis you can't physically rotate the wheel and repeatedly flicking the wheel to the left or right is a lot slower and more strenuous compared to rotating the wheel on the vertical axis. As such, if you hold a horizontal tilt the mouse will just repeatedly send the horizontal scroll signal so as to emulate flicking at a pace similar to the number of ticks people can output when scrolling vertically. But to still let the user send precisely controlled singular flicks, they use this delay to distinguish between "flicks" and "holds". 

And unfortunately that probably means there is no easy way to disable this...
